Emby.Naming/TV/SeriesPathParser.cs

```diff
@@ -0,0 +1,64 @@
+using System.Globalization;
+using Emby.Naming.Common;
+
+namespace Emby.Naming.TV
+{
+    /// <summary>
+    /// Used to parse information about series from paths containing more information that only the series name.
+    /// Uses the same regular expressions as the EpisodePathParser but have different success criteria.
+    /// </summary>
+    public static class SeriesPathParser
+    {
+        /// <summary>
+        /// Parses information about series from path.
+        /// </summary>
+        /// <param name="options"><see cref="NamingOptions"/> object containing EpisodeExpressions and MultipleEpisodeExpressions.</param>
+        /// <param name="path">Path.</param>
+        /// <returns>Returns <see cref="SeriesPathParserResult"/> object.</returns>
+        public static SeriesPathParserResult Parse(NamingOptions options, string path)
+        {
+            SeriesPathParserResult? result = null;
+
+            foreach (var expression in options.EpisodeExpressions)
+            {
+                var currentResult = Parse(path, expression);
+                if (currentResult.Success)
+                {
+                    result = currentResult;
+                    break;
+                }
+            }
+
+            if (result != null)
+            {
+                if (!string.IsNullOrEmpty(result.SeriesName))
+                {
+                    result.SeriesName = result.SeriesName
+                        .Trim()
+                        .Trim('_', '.', '-')
+                        .Trim();
```

Comment:
Minor optimization thing (but given a few thousand items might add up), I wonder how often do we encounter special unicode whitespace chars? Would this `.Trim(' ', '_', '.', '-')` not cover something like 99% of cases?

Right now we're potentially creating 3 new strings for every file. The good thing about `.Trim()` is that it trims _all_ whitespace chars, but I can't help but wonder if it's even necessary.

Reply:
Probably good enough with  `.Trim(' ', '_', '.', '-')` yes

Reply:
Fixed